core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMaterializedViewDefinition.java

```diff
@@ -33,6 +34,16 @@
     private final Optional<String> comment;
     private final Optional<String> owner;
     private final Map<String, Object> properties;
+    /**
+     * Set of versioning columns that engine will provide during MV refresh.
+     * MV storage table should provide same set of columns with matching names.
+     * Empty if MV does not support versioning.
+     */
+    private final Optional<Set<Column>> versioningColumns;
```

Comment:
How does the versioning column exactly mean? Are these columns used to identify if the corresponding table is changed or not? Will these columns be given by a user in Create-MV DDL?

Reply:
When MV target connector supports versioning, then during incremental refresh target connector should INSERT OVERRIDE groups specified by `versioningColumns`.
This can INSERT OVERWRITE on row level (aka row-level update) or for example INSERT OVERWRITE on partition level

> Will these columns be given by a user in Create-MV DDL?

No, these will be handled internally by engine during refresh. These won't be visible during MV scan unless user specifically selects them